src/features/staking/subcomponents/ApproveSuccess.tsx

```diff
@@ -1,79 +1,69 @@
-import { Dispatch, FC } from "react";
+import { FC } from "react";
 import { IoMdOpen } from "react-icons/io";
 import greenCheck from "../../../assets/check-green.svg";
-import { etherscanLink } from "../../../utils/constants";
-import { StatusStaking } from "../types/StakingTypes";
+import { StakeOrUnstake, WagmiLoadingStatus } from "../types/StakingTypes";
+import { etherscanLink } from "../utils/helpers";
 
 interface ApproveSuccessProps {
-  statusStaking: StatusStaking;
-  setStatusStaking: Dispatch<StatusStaking>;
-  amountApproved?: string;
-  amountStaked?: string;
+  stakeOrUnstake: "stake" | "unstake";
+  statusUnstake: WagmiLoadingStatus;
+  amount: string;
   chainId: number;
-  transactionHashApprove?: string | undefined;
   transactionHashStake?: string | undefined;
+  transactionHashUnstake?: string | undefined;
 }
 
 const ApproveSuccess: FC<ApproveSuccessProps> = ({
-  statusStaking,
-  setStatusStaking,
-  amountApproved,
-  amountStaked,
+  stakeOrUnstake,
+  amount,
   chainId,
-  transactionHashApprove,
   transactionHashStake,
+  transactionHashUnstake,
 }) => {
-  const handleCloseMessage = () => {
-    if (statusStaking === "approved") {
-      setStatusStaking("readyToStake");
-    } else if (statusStaking === "success") {
-      setStatusStaking("unapproved");
+  const handleMessage = () => {
+    if (stakeOrUnstake === StakeOrUnstake.STAKE) {
+      return "You've successfully staked";
+    } else {
+      return "You've successfully unstaked";
     }
   };
+  const message = handleMessage();
```

Comment:
(This also applies to `handleBlockExplorerLink` below)

You don't need to define an intermediate funciton here, you can simply do `const message = stakeOrUnstake === StakeOrUnstake.STAKE ? "Successfully staked" : "Successfully Unstaked";`

Equally you should either accept the `StakeOrUnstake` enum as the prop type or compare with the value directly.